include/lbann/callbacks/export_onnx.hpp

```diff
@@ -0,0 +1,97 @@
+////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2014-2019, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory.
+// Written by the LBANN Research Team (B. Van Essen, et al.) listed in
+// the CONTRIBUTORS file. <lbann-dev@llnl.gov>
+//
+// LLNL-CODE-697807.
+// All rights reserved.
+//
+// This file is part of LBANN: Livermore Big Artificial Neural Network
+// Toolkit. For details, see http://software.llnl.gov/LBANN or
+// https://github.com/LLNL/LBANN.
+//
+// Licensed under the Apache License, Version 2.0 (the "Licensee"); you
+// may not use this file except in compliance with the License.  You may
+// obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the license.
+//
+// export_onnx .hpp .cpp - Exports trained model to onnx format
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef LBANN_CALLBACKS_EXPORT_ONNX_HPP_INCLUDED
+#define LBANN_CALLBACKS_EXPORT_ONNX_HPP_INCLUDED
+
+#ifdef LBANN_HAS_ONNX
+#include <onnx/onnx_pb.h>
+#endif // LBANN_HAS_ONNX
+#include "lbann/callbacks/callback.hpp"
+#include <google/protobuf/message.h>
+#include <lbann/base.hpp>
+#include <iostream>
+#include <memory>
+#include <vector>
+
+namespace lbann {
+namespace callback {
+
+/** @class export_onnx
+ *  @brief Callback to export a trained model to onnx format
+ */
+class export_onnx : public callback_base {
+
+public:
+  /** @brief export_onnx Constructor.
+   *  @param output_file Output filename (default = lbann_output.onnx)
+   *  @param print_debug_string Option to print debug string to stdout
+   */
+  export_onnx(bool print_debug_string = false,
+              std::string output_file = "lbann_output.onnx");
+
+  /** @brief Copy constructor */
```

Comment:
```suggestion
  /** @brief Copy interface */
```
But also, why is it not copyable? Protobuf messages are copyable. (FWIW, I don't think it makes sense to copy a callback ever, but somebody thought it made sense to add the `copy()` function. Since this object actually is copyable, and makes as much sense to copy as any other callback, we should support this function as well as any other callback...)